src/features/world/ui/SpecialEventModalContent.tsx

```diff
@@ -149,43 +158,53 @@ export const SpecialEventBumpkin: React.FC<{ onClose: () => void }> = ({
     );
   }
 
-  if (showWallet) {
+  if (reward) {
     return (
       <Panel>
-        <div className="p-2">
-          <Label icon={walletIcon} type="default" className="mb-2">
-            Wallet Required
-          </Label>
-          <p className="text-sm mb-2">
-            A Web3 wallet is required to claim this airdrop
-          </p>
-          <p className="text-xs mb-2">
-            Airdrops are handled externally and may take a few days to arrive.
-          </p>
-        </div>
-        <Button>Continue</Button>
+        <ClaimReward reward={reward} onClose={() => setReward(undefined)} />
       </Panel>
     );
   }
 
-  if (reward) {
+  // isEligible should already be checked by the parent component but just in
+  // case it was missed, let's check it here as well
+  if (event.isEligible) {
```

Comment:
Should this be `!event.isEligible`?

I currently get this when loading the NPC to start the mission.

<img width="556" alt="Screenshot 2024-02-16 at 8 29 01 am" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/5f6415c7-5c2d-42cc-aabe-64c3ffcc4234">